include/RouteMap.h

```diff
@@ -305,57 +324,119 @@ struct RouteMapPosition {
+    double MaxTrueWindKnots;
+    // The calculated route will avoid a path where the gust wind is above this value in knots.
+    double MaxApparentWindKnots;
+
+    // The calculated route will avoid swells larger than this value in meters.
+    // If the grib data does not contain swell information, the maximum swell value is ignored.
+    // If there is no route within the maximum swell value, the route calculation will fail.
+    double MaxSwellMeters;
+    // The calculated route will not go beyond this latitude, as an absolute value.
+    // If the starting or destination position is beyond this latitude, the route calculation will fail.
+    double MaxLatitude;
+    // The penalty time to tack the boat, in seconds.
+    // The penalty time is added to the route calculation for each tack.
+    double TackingTime;
+    // Balance the influence of the wind and the ocean current on the route calculation.
+    double WindVSCurrent;
+    // The minimum safety distance to land, in nautical miles.
+    // The calculated route will avoid land within this distance.
     double SafetyMarginLand;
 
     bool AvoidCycloneTracks;
-    int CycloneMonths, CycloneDays;
+    // Avoid cyclone tracks within ( 30*CycloneMonths + CycloneDays ) days of climatology data.
+    int CycloneMonths;
+    int CycloneDays;
 
     bool UseGrib;
     enum ClimatologyDataType {DISABLED, CURRENTS_ONLY, CUMULATIVE_MAP, CUMULATIVE_MINUS_CALMS, MOST_LIKELY, AVERAGE};
     enum ClimatologyDataType ClimatologyType;
     bool AllowDataDeficient;
-    double WindStrength; // wind speed multiplier
+    double WindStrength; // wind speed multiplier. 1.0 is 100% of the wind speed in the grib.
+
+    // If true, the route calculation will avoid land, outside the SafetyMarginLand.
+    bool DetectLand;
+    // If true, the route calculation will avoid exclusion boundaries.
+    bool DetectBoundary;
+    // If true and grib data contains ocean currents, the route calculation will use ocean current data.
+    bool Currents;
+    // If true, avoid polar dead zones.
+    // If false, avoid upwind course (polar angle too low) or downwind no-go zone (polar angle too high).
+    bool OptimizeTacking;
+    bool InvertedRegions;
+    bool Anchoring;
+
+    // Do not go below this minimum True Wind angle at each step of the route calculation.
+    // The default value is 0 degrees.
+    double FromDegree;
+    // Do not go above this maximum True Wind angle at each step of the route calculation.
+    // The default value is 180 degrees.
+    double ToDegree;
+    // The angular resolution at each step of the route calculation, in degrees.
+    // Lower values provide finer resolution but increase computation time.
+    // Higher values provide coarser resolution, but faster computation time.
+    // The allowed range of resolution is from 0.1 to 60 degrees.
+    // The default value is 5 degrees.
+    double ByDegrees;
```

Comment:
https://github.com/seandepagnier/weather_routing_pi/blob/05090737c2460a367f41936db9978267fcc97572/src/RouteMap.cpp#L2542-L2547

Reply:
I believe so that the degree stops don't change during a computation, but it has been some time since I wrote this.